### ISSUE TITLE
Unable to terminate a suspended workflow

### ISSUE THREAD

**Issue by MrSaints** [argoproj/argo#1407]
**Is this a BUG REPORT or FEATURE REQUEST?**: Bug report

**What happened**: I submitted a workflow with a task pointed to a suspend template. The workflow reaches a suspended state. All good. However, when I try to terminate the suspended workflow, nothing happens. It seems that the workflow is unable to transition from the suspended state to a terminated state.

**What you expected to happen**: Terminate should override suspend.

**How to reproduce it (as minimally and precisely as possible)**:

1. Start a workflow with a DAG containing a task using a suspend template (e.g. https://github.com/argoproj/argo/blob/master/examples/suspend-template.yaml)
2. Run `argo terminate` on the workflow
3. Check the workflow

**Anything else we need to know?**:

**Environment**:
- Argo version:

```
argo: v2.2.1
  BuildDate: 2018-10-11T16:25:59Z
  GitCommit: 3b52b26190163d1f72f3aef1a39f9f291378dafb
  GitTreeState: clean
  GitTag: v2.2.1
  GoVersion: go1.10.3
  Compiler: gc
  Platform: darwin/amd64
```

```
$ argo version
```
- Kubernetes version : v2.3.0

```
$ kubectl version -o yaml
```

**Other debugging information (if applicable)**:
- workflow result:
```
$ argo get <workflowname>
```
- executor logs:
```
$ kubectl logs <failedpodname> -c init
$ kubectl logs <failedpodname> -c wait
```
- workflow-controller logs:
```
$ kubectl logs -n kube-system $(kubectl get pods -l app=workflow-controller -n kube-system -o name)
```
There are no obvious logs.

`Workflow '<omitted>' terminated`

**Comment by MrSaints** [comment]
Having a look at the code, I believe this is because termination works by setting the timeout to 0 which has no impact on a suspended workflow.

**Comment by sarabala1979** [comment]
Terminate should work irrespective of status suspend or running

**Comment by mark9white** [comment]
I may have time to look at this shortly. @sarabala1979 if you've already got any thoughts on an approach here it would be useful to hear them.

**Comment by MrSaints** [comment]
I believe this may be resolved in v2.5.x. I successfully terminated a suspended workflow via the UI.

**Comment by mark9white** [comment]
I believe this is fixed by #1704

**Comment by simster7** [comment]
Closing, thanks guys!
